### ISSUE TITLE
Bug in imageByReplacingColor

### ISSUE THREAD

**Issue by stolpovsky** [bennyguitar/BGUtilities#1]
I was replacing color in UIImage, and no matter what replacement color I used, the resulting image would have badColor replaced with transparent instead.

It's easy to see why -- your scaling is off. The values of your aRGBA, bRGBA arrays are within [0, 1]. But data[i] is [0x00, 0xff]. To compare data[i] to aRGBA, you are dividing data[i] by 255.0.

```
    redIn = (double)data[i]/255.0;
```

That is correct. Then you do the comparison:

```
    if(redIn == [aRGBA[0] floatValue] <...>)
```

That is correct, too. But then you set a [0, 1] number to data[i] which expects [0x00, 0xff]:

```
        data[i] = [bRGBA[0] floatValue];
```

That is wrong. Should be scaled by 255., something like this:

```
255. * (float) [bRGBA[0] floatValue];
```

**Comment by stolpovsky** [comment]
Oh, and you have the same problem in 

```
imageByReplacingColorsWithinDistance
```

I am using 1.4.0. Thanks.

**Comment by bennyguitar** [comment]
Fixed - now on 1.5.0.
